src/meta.rs

```diff
@@ -58,7 +62,12 @@ impl Meta {
             "{}-extracted",
             self.resource_path.file_name().unwrap().to_str().unwrap()
         );
-        self.resource_path.parent().unwrap().join(dirname)
+
+        if let Some(extraction_path) = &self.extraction_path {
+            self.resource_path.parent().unwrap().join(extraction_path)
+        } else {
+            self.resource_path.parent().unwrap().join(dirname)
```

Comment:
Nit: can you move the `dirname` declaration above to this block since it's only needed here?